clip.md

```diff
@@ -0,0 +1,443 @@
+    ~> call CLIP_DOG . digs CLIP_ILK ( #if LOT_FINAL ==Wad wad(0) #then TAB_FINAL +Rad Wad2Rad(OWE_FINAL) #else Wad2Rad(OWE_FINAL) #fi )
+    ~> (#if LOT_FINAL ==Wad wad(0) #then Clip . remove ID #else ( #if TAB_FINAL ==Rad rad(0) #then call CLIP_VAT . flux CLIP_ILK THIS USR LOT_FINAL #else . #fi )  #fi)
+    ~> (#if LOT_FINAL ==Wad wad(0) #then .                #else ( #if TAB_FINAL ==Rad rad(0) #then Clip . remove ID #else .                                 #fi )  #fi)
+    ~> (#if LOT_FINAL ==Wad wad(0) #then false            #else ( #if TAB_FINAL ==Rad rad(0) #then false            #else true                              #fi )  #fi)
+    ~> TAB
+    ~> SLICE_INITIAL
+    ~> OWE_FINAL
+    ~> PRICE
+    ~> DONE
+    ~> emitTake ID MAX PRICE OWE_FINAL TAB_FINAL LOT_FINAL USR
+    )( LOT -Wad SLICE_FINAL )
+    )( TAB -Rad Wad2Rad(OWE_FINAL) )
+    )( #if (OWE_INITIAL >Wad Rad2Wad(TAB)) #then OWE_FINAL /Wad PRICE #else ( #if (OWE_INITIAL <Wad Rad2Wad(TAB) andBool SLICE_INITIAL <Wad LOT ) #then (#if ( ( Rad2Wad(TAB) -Wad OWE_INITIAL ) <Wad Ray2Wad(CLIP_CHOST) ) #then ( OWE_FINAL /Wad PRICE ) #else SLICE_INITIAL #fi ) #else SLICE_INITIAL #fi ) #fi )
+    )( #if (OWE_INITIAL >Wad Rad2Wad(TAB)) #then Rad2Wad(TAB) #else ( #if (OWE_INITIAL <Wad Rad2Wad(TAB) andBool SLICE_INITIAL <Wad LOT ) #then (#if ( ( Rad2Wad(TAB) -Wad OWE_INITIAL ) <Wad Ray2Wad(CLIP_CHOST) ) #then (Rad2Wad(TAB) -Wad Ray2Wad(CLIP_CHOST) ) #else OWE_INITIAL #fi ) #else OWE_INITIAL #fi ) #fi )
+    )( SLICE_INITIAL *Wad Ray2Wad(PRICE) )
+    )( minWad(LOT, AMT) )
+    )( ( (NOW -Int TIC ) >Int CLIP_TAIL ) orBool ( (PRICE /Ray TOP) <Ray CLIP_CUSP ) )
+    )( Abacus CLIP_ABACUS . price TOP (NOW -Int TIC) )
+    ... </k>
+    <clip-sales> ... ID |-> ClipSale( ... tab: TAB, lot: LOT, usr: USR, tic: TIC, top: TOP ) ... </clip-sales>
+    <clip-chost>    CLIP_CHOST    </clip-chost>
+    <clip-ilk>      CLIP_ILK      </clip-ilk>
+    <clip-vat>      CLIP_VAT      </clip-vat>
+    <clip-dog>      CLIP_DOG      </clip-dog>
+    <clip-vow>      CLIP_VOW      </clip-vow>
+    <clip-tail>     CLIP_TAIL     </clip-tail>
+    <clip-cusp>     CLIP_CUSP     </clip-cusp>
+    <clip-abacus>   CLIP_ABACUS   </clip-abacus>
+    <clip-stopped>  CLIP_STOPPED  </clip-stopped>
+    <current-time> NOW </current-time>
+    <this>       THIS       </this>
+    <msg-sender> MSG_SENDER </msg-sender>
+    requires CLIP_STOPPED <ClipStop noNewKickOrRedoOrTake
+    andBool USR =/=K 0:Address
+
+    rule <k> UPDATE:Bool ~> TAB:Rad ~> SLICE_INITIAL:Wad ~> OWE_INITIAL:Wad ~> PRICE:Ray ~> DONE:Bool ~> emitTake ID MAX PRICE OWE_FINAL TAB_FINAL LOT_FINAL USR => emitTake ID MAX PRICE OWE_FINAL TAB_FINAL LOT_FINAL USR ... </k>
+    <clip-sales> ... ID |-> ClipSale( ... tab: (TAB => #if UPDATE #then TAB_FINAL #else TAB #fi), lot:(LOT => #if UPDATE #then LOT_FINAL #else LOT #fi) ) ... </clip-sales>
+    <clip-chost> CLIP_CHOST </clip-chost>
+    requires DONE
+    andBool MAX >=Ray PRICE
```

Comment:
Not sure I fully understand what's going on here, but if `DONE` is true, the function reverts, whereas `MAX >=Ray PRICE` is a non-reversion condition...I suspect you want to negate `DONE` in this context.

Reply:
That it correct, added `notBool` before `DONE` and now the condition is negated.